tests/roundtrip.rs

```diff
@@ -174,9 +173,8 @@ mod tests {
         let df = ctx.sql(sql).await?;
         let plan1 = df.to_logical_plan()?;
         let proto = to_substrait_plan(&plan1)?;
-
-        let df = from_substrait_plan(&mut ctx, &proto).await?;
-        let plan2 = df.to_logical_plan()?;
+        let plan2 = from_substrait_plan(&mut ctx, &proto).await?;
+        let plan2 = ctx.optimize(&plan2)?;
```

Comment:
Clarification question. Why do we need to call `optimize()` here?

Reply:
Before the changes in this PR, we were receiving a `DataFrame` and then calling `DataFrame::to_logical_plan`, which performed the optimization step, so the existing tests were expecting an optimized plan.

Reply:
Ah I see. Thank you for the explanation!